src/core/server/models/comment/revision.ts

```diff
@@ -45,6 +54,21 @@ export interface RevisionMetadata {
    * without a warning.
    */
   nudge?: boolean;
+
+  /**
+   * externalModeration is any details about if and when this comment revision
+   * was analyzed by an external moderation phase.
+   */
+  externalModeration?: {
+    name: string;
+    analyzedAt: Date;
+    status?: GQLCOMMENT_STATUS;
+    tags?: GQLTAG[];
+    actions?: {
+      type?: ACTION_TYPE;
+      reason?: FLAG_REASON;
+    }[];
```

Comment:
May I suggest nesting this one level deeper under `result` or something similar to distinguish it from the top level moderation properties?

Reply:
Unsure what you mean here. The other results also do this same pattern? There's:

- perspective
- wordList
- akismet
- externalModeration

You want a results inside the externalModeration type? Or a results with an externalModeration inside?

If the former, then we should add a results subtype to perspective, wordList, akismet, etc.
If the latter, then we need to move akismet, wordList, perspective, etc inside results too.

Also, on the topic of "distinguish it from the top level moderation properties", that is kind of the point of this whole PR after talking with Sam/Andrew. 

The clients that use this want to treat the external moderation phases as 1st class mod phases. We add a lot of visibility in the mod-details and even merge them all under the same "Details" in the mod cards. Even going as far as to mark the mod card "moderated by" to show <external_mod_phase> instead of "System".

I don't think treating these as less than "top level" is accurate anymore.

Reply:
Ah, after discussion, you meant `add a results subtype` to `externalModeration`. All good, it will be done!